src/components/main/ItemsGrid.js

```diff
@@ -62,13 +105,52 @@ class ItemsGrid extends Component {
   };
 
   render() {
-    const { title, itemSearch } = this.props;
+    const { classes, items, title, itemSearch } = this.props;
+    const { page, itemsPerPage } = this.state;
+
+    const pagesCount = Math.ceil(items.size / itemsPerPage);
+
+    const start = (page - 1) * itemsPerPage;
+    const end = start + itemsPerPage;
+    const itemsInPage = items.slice(start, end);
+
     return (
       <div>
         <TableToolbar tableTitle={title} itemSearchInput={itemSearch?.input} />
         <Grid container spacing={1}>
-          {this.renderItems()}
+          {this.renderItems(itemsInPage)}
         </Grid>
+        <Box
+          display="flex"
+          justifyContent="center"
+          className={classes.paginationControls}
+        >
+          <Box className={classes.itemsPerPageSelect}>
```

Comment:
I would put this select on the top right. Otherwise it looks a bit weird here. The other option would be to have a proper footer division (e.g. a top border) that indicates that the whole bottom area is a "pagination settings" area. Or maybe change the select style to be contained, so that it looks less like it's just floating in the corner.